### DISCUSSION TITLE
Bootstrapping the complete build

### DISCUSSION THREAD

**Discussion by petersasi** [petersasi/deluge2-win-build#22] [Ideas]
Hi @mhertz as you can see I am trying to separate the several topics we have under discussion to separate threads for easier to read threads. 😅
This is a new one that I got envious of from  few projects that do not need a complicated environment setup to start:
https://github.com/petersasi/deluge2-win-build/blob/master/00-bootstrap.cmd

1. Of course this will also need us to keep everything under the single folder where this is started, thus as you can see I have moved MSYS2 there as well as the git repo clone folder.
2. This comes before https://github.com/petersasi/deluge2-win-build/blob/master/install_components.cmd, and I plan to invoke all the necessary step in the necessary order from it, similar to e.g. the one step auto install of https://docs.pi-hole.net/main/basic-install/ 
`curl -sSL -O https://github.com/petersasi/deluge2-win-build/raw/master/00-bootstrap.cmd && 00-bootstrap.cmd`

**Comment by mhertz** [comment]
Also great idea, makes sense :) Also containing everything in single self-choosen folder as you stated, is really nice imho. 

I would personally in bootstrapping script use curl to get initial setup "cloned", instead of needing git as requirement. 

Off-topic sorry, but note gvsbuild has issues when building in folder with around +25 chars or something(e.g in Documents or Downloads standard win10 folders deluge2 subfolder), which I had a hell of a time finding workaround for, which you can see in my posted ticket about it on gvsbuild github(I closed it btw to not clutter and as issue gone since found workaround, just if interested later when coming to it.

**Reply by mhertz** [reply]
OK I'm an idiot of-course, please forget my extremely clever comment about what I would do instead lol 🤣

**Comment by petersasi** [comment]
Yes, I observed your "fight" with path length and am grateful you resolved that - although had not had time to incorporate those changes either. 😅

**Comment by petersasi** [comment]
Okay, so in the end I stopped messing up my work computer and downloaded a W10 dev VM from MS to test this in. It will not test the MSVC installation, but at least should be able to fix up the rest. :)

**Comment by mhertz** [comment]
Yeah that's pretty cool msft provides that, and I looked at them too last time I by mistake deleted my win10 VM's because stupidly remembered wrong my shell-aliases for some of it :)
